src/main/java/io/cdap/plugin/cdc/source/sqlserver/CTInputDStream.java

```diff
@@ -49,17 +50,43 @@
 public class CTInputDStream extends InputDStream<StructuredRecord> {
   private static final Logger LOG = LoggerFactory.getLogger(CTInputDStream.class);
   private final JdbcRDD.ConnectionFactory connectionFactory;
+  private final long maxRetrySeconds;
   private long trackingOffset;
+  private long failureStartTime;
+  private boolean isFailing;
 
-  CTInputDStream(StreamingContext ssc, JdbcRDD.ConnectionFactory connectionFactory) {
+  CTInputDStream(StreamingContext ssc, JdbcRDD.ConnectionFactory connectionFactory,
+                 long maxRetrySeconds) {
     super(ssc, ClassTag$.MODULE$.apply(StructuredRecord.class));
     this.connectionFactory = connectionFactory;
+    this.maxRetrySeconds = maxRetrySeconds;
     // if not current tracking version is given initialize it to 0
     trackingOffset = 0;
   }
 
   @Override
   public Option<RDD<StructuredRecord>> compute(Time validTime) {
+    try {
+      return doCompute(validTime);
+    } catch (Exception e) {
+      if (isFailing && maxRetrySeconds >= 0) {
+        if (maxRetrySeconds == 0 || maxRetrySeconds > (nowInSeconds() - failureStartTime)) {
```

Comment:
Hi,
we should exit when current waiting time exceeds maxRetrySeconds
```suggestion
        if (maxRetrySeconds == 0 || maxRetrySeconds < (nowInSeconds() - failureStartTime)) {
```

Also the code "throw e;" doesn't finish the pipeline execution. 
I've tried to add "super.ssc().stop(true);" line. Which actually stops pipeline execution, however, the pipeline finishes with "Succeeded" status which is wrong.